### PR TITLE
Sr865 driver with buffered acquisition support

### PR DESCRIPTION
**Pull request by sohailc**
I have added buffered reading support to the SR865/SR860 driver. An example notebook with usage is added.

### CODE REVIEW

File: qcodes/instrument_drivers/stanford_research/SR86x.py [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+        n = np.log2(max_rate / capture_rate_hz)
+        n_round = int(round(n))
+
+        if not 0 <= n_round <= 20:
+            raise ValueError("The chosen frequency is invalid. Please consult the SR860 manual at page 136. The maximum"
+                             " capture rate is {}".format(max_rate))
+
+        nearest_valid_rate = max_rate / 2 ** n_round
+        if abs(capture_rate_hz - nearest_valid_rate) > 1:
+            print("Warning: Setting capture rate to {:.5} Hz".format(nearest_valid_rate))
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/stanford_research/SR86x.py]: I would log this too

File: qcodes/instrument_drivers/stanford_research/SR86x.py [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+    def __init__(self, name, address, reset=False, **kwargs):
+        super().__init__(name, address, terminator='\n', **kwargs)
+        # Reference commands
+        self.add_parameter(name='frequency',
+                           label='Frequency',
+                           unit='Hz',
+                           get_cmd='FREQ?',
+                           set_cmd='FREQ {}',
+                           get_parser=float,
+                           vals=Numbers(min_value=1e-3, max_value=2.5e6))
```
Comment (jenshnielsen) [inline qcodes/instrument_drivers/stanford_research/SR86x.py]: I know this is from the old driver and you didn't write this. Never the less it seems strange to me that the max value is 2.5 MHz. As far as i can see this should be 5 KHz for the SR860 and 4 MHz for the SR865

**Comment by sohailc** [comment]
Thanks @jenshnielsen for your comments. I will look at them now and update the driver

File: qcodes/instrument_drivers/stanford_research/SR86x.py [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+        self._capture_data = capture_data
+
+        data_len = len(capture_data)
+        self.shape = (data_len,)
+        self.setpoint_units = ('',)
+        self.setpoint_names = ('sample_nr',)
+        self.setpoint_labels = ('Sample number',)
+        self.setpoints = (tuple(np.arange(0, data_len)),)
+
+    def get_raw(self) ->None:
```
[the remark this answers is not in the document]
Comment (sohailc) [reply qcodes/instrument_drivers/stanford_research/SR86x.py]: Thank for you comments Jens. I will update accordingly, test and merge (if everything works)

**Pull request merged**